upgrade/v100to101/upgrade.go

```diff
@@ -51,8 +48,11 @@ func UpgradeInstanceManagerPods(namespace string, lhClient *lhclientset.Clientse
 			return errors.Wrapf(err, upgradeLogPrefix+"failed to find pod for instance manager %v during the instance managers pods upgrade", im.Name)
 		}
 		for _, pod := range imPodsList.Items {
-			if err = upgradeInstanceMangerPodLabel(&pod, &im, kubeClient, namespace); err != nil {
-				return err
+			newInstanceManagerLabels := types.GetInstanceManagerLabels(im.Spec.NodeID, im.Spec.Image, im.Spec.Type)
+			if pod.Labels[types.GetLonghornLabelKey(types.LonghornLabelInstanceManagerImage)] != newInstanceManagerLabels[types.GetLonghornLabelKey(types.LonghornLabelInstanceManagerImage)] {
```

Comment:
Consider the following case:
The 1st map doesn't contain the key. The 2nd map contains the key but the value somehow is empty. 

Hence I think it's better to use `reflect.DeepEqual()`. 

Same below.

Reply:
Thank you, fixed!

The second map will always contains the key though. Because we updated IM object in v700tov800 before getting here.

Reply:
What if we're going to have more labels for the pod in the future? Then this DeepEquals will result in overriding other labels.

Reply:
In the case when we have more labels do we also update the function `types.GetInstanceManagerLabels()` to include them? If so, then I think it should be ok

Reply:
What if other controllers add unknown labels to the pod?

Reply:
Thanks. Went back to the previous version. lol. Since it is safe to assume that the value is non-empty.